### PR TITLE
Update docker configuration

### LINKED ISSUE #421: Docker Environment variables for configuration
Environment variables for config are not documented. See https://github.com/PostgREST/postgrest/pull/1691.

### PR DESCRIPTION
**Pull request by laurenceisla**
Closes #421.

Should be merged after #422 due to the reorganization of the configuration page.

### CODE REVIEW

File: configuration.rst [hunk 1]
```diff
@@ -73,17 +80,6 @@ To reload the configuration without restarting the PostgREST server send a SIGUS
...
-.. note::
-
-   To refresh the cache in docker:
-
-   .. code:: bash
-
-     docker kill -s SIGUSR2 <container>
-
-     # or in docker-compose
-     docker-compose kill -s SIGUSR2 <service>
```
Comment (laurenceisla) [inline configuration.rst:85]: According to these comments https://github.com/PostgREST/postgrest/pull/1691#discussion_r542065652 this isn't possible in Docker now that no config file is used. After #422 is merged, there should be a reference to In-db configuration reloading instead.

Comment (steve-chavez) [reply configuration.rst:85]: Yeah, true. This should be removed.

File: install.rst [hunk 2]
```diff
@@ -99,19 +99,7 @@ You can get the `official PostgREST Docker image <https://hub.docker.com/r/postg
 
   docker pull postgrest/postgrest
 
-The image consults an internal ``/etc/postgrest.conf`` file. To customize this file you can either mount a replacement configuration file into the container, or use environment variables. The environment variables will be interpolated into the default config file.
```
Comment (steve-chavez) [inline install.rst:102]: This breaking change should be listed in Upcoming.

Comment (steve-chavez) [reply install.rst:102]: Btw, why the config file inside docker don't work anymore?

Comment (steve-chavez) [reply install.rst:102]: I see that the Nix docker script did include the [config file](https://github.com/PostgREST/postgrest/pull/1522/files#diff-efc6bf6ccfbaf2c72cc63bcb3b11e5871273bbd23a7bf63e5e790f16aca835edR18) but at some point it was removed([current](https://github.com/PostgREST/postgrest/blob/main/nix/tools/docker/default.nix)).

Comment (steve-chavez) [reply install.rst:102]: Ah, it was removed in https://github.com/PostgREST/postgrest/pull/1691/. This should be noted in the CHANGELOG.

**Review by steve-chavez — APPROVED** [review]

**Pull request merged**